### ISSUE TITLE
Carbon Web Component: Footer - Language selector, issues on screen reader (JAWS)

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/carbon-for-ibm-dotcom#4889]
## Detailed description

Issue:
1. There is no label
2. Screen reader announces this as a list box. Hence the user does not know that this is an Edit combo.
3. When we navigate within the combo box using up/down arrow keys, Screen reader does not announce anything. Hence Unable to select the preferred option 

In React staging, The language selector control is not identified by jaws in arrow key navigation. An issue is already opened on the same and is being tracked.

Expected:
1. Provide appropriate label
2. If this is an editable combo box, Screen reader should convey the "edit combo" role to the user 
3. Screen reader Users should be able to navigate and make the selection within the combo box.

**Comment by IgnacioBecerra** [comment]
@Praveen-sr I'm able to navigate through the combo box and the item list with VoiceOver properly. I know this is a JAWS issue, but I'm under the impression it would also work with it. Would you be able to re-test this? If the issue still persists, please let me know!

For the moment I opened a PR (#4914) that addresses the `edit combo` issue, as well as adding an aria label to the clear button.

cc: @RobertaJHahn

**Comment by RobertaJHahn** [comment]
Reviewed with Ram in QA meeting

**Comment by Praveen-sr** [comment]
@IgnacioBecerra Original defect fixed but few regression issues found.

Now JAWS identifies the list box Role in up/down arrow navigation. But the following issues are found.
1. In up/down arrow navigation, the list box role is announced without any label. Where as, in tab/shift tab navigation, the same is announced as "Edit combo box". This confuses the user.
2. There is a clear button present after this control which is not identified in up/down arrow navigation. It is identified only in tab/shift tab navigation. And there is no label for this button as well.
3. After providing an input. Or without providing any input, when we navigate through the dropdown options using up/down arrow keys, JAWS announces the options in an Incomprehensible way. Hence the user does not understand which option /Language it is.

**Comment by IgnacioBecerra** [comment]
@Praveen-sr Thank you, there is an open PR addressing the first two issues.

Could you expand upon the “incomprehensible way” when reading the dropdown options? Though I'm thinking this may be a bug with JAWS itself, due to the way it handles input fields.

I found two issues about that, [one within carbon components](https://github.com/carbon-design-system/carbon/issues/3804) that links to the [JAWS issue itself,](https://github.com/FreedomScientific/VFO-standards-support/issues/199#issuecomment-521641212) since I know that using tab navigation the voiceover works properly.

cc: @RobertaJHahn

**Comment by Praveen-sr** [comment]
@IgnacioBecerra 
1. We have retested this defect in jaws 2020 and also in jaws 2019 and found the same behavior. To verify whether this is specific to jaws, we also checked with NVDA which is another windows Screen reader and found the same behavior there too.
2. Here is screen shoot of JAWS reading the drop down options :

![image.png](https://images.zenhubusercontent.com/5e21a7b470118a498429a20e/1fd7f54c-6cae-4d34-b24d-5519f0cf1de7)

CC @RobertaJHahn

**Comment by RobertaJHahn** [comment]
@IgnacioBecerra Adding to the engineering issues refinement meeting agenda, Jan 25.  Sounds like it might be good to get input from other engineers on this one.  

cc:  @Praveen-sr

**Comment by RobertaJHahn** [comment]
Next steps:  1)  Anna and Ariella will install JAWS to test on our end.  2)  If still and issue create a bug against the carbon `combo-box` component.

**Comment by ariellalgilmore** [comment]
update: we were not granted access to get windows VM on our computers, so do not have the ability to test JAWS

**Comment by IgnacioBecerra** [comment]
Update: We were able to get another version of VM and JAWS installed in our systems.

After some testing, I managed to find a solution. Since this issue comes from the components the Language Selector extends from, `ComboBox` and `Dropdown`, from carbon-web-components, it will have to be a contribution to the repo.

I opened a PR (https://github.com/carbon-design-system/carbon-web-components/pull/580) to address this issue from the source.

cc: @RobertaJHahn

**Comment by Praveen-sr** [comment]
@IgnacioBecerra Ram has tested the "edit combo" in both chrome and FF.
The behavior you mentioned is found in Chrome. 
In FF, the Edit combo role is not identified by jaws. And also the selected value is not announced after making a selection. hence the user would not know the currently selected option.
Regarding the jaws announcement of the input value along with each option in chrome, I think that is tolerable as long as we can achieve the same behavior in FF as in chrome. I would mark that as low sev or even a best practice.

**Comment by kennylam** [comment]
Reopening as this issue still exists.
